src/Validation/Constraint/HasClaim.php

```diff
@@ -0,0 +1,30 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation\Constraint;
+
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\UnencryptedToken;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\ConstraintViolation;
+
+final class HasClaim implements Constraint
+{
+    /** @param non-empty-string $claim */
+    public function __construct(private readonly string $claim)
+    {
```

Comment:
We don't want Registered Claims to be checked here, see

https://github.com/lcobucci/jwt/blob/d6d30cde835c9399219df8496b4993cb1abd2b77/src/Validation/Constraint/HasClaimWithValue.php#L18-L20